index.js

```diff
@@ -1,294 +1,155 @@
+			res.send(errmessage);
+		} else {
+			res.send(post);
 		}
-	}
-	res.send(db.posts[post_number]);
+	})
 });
 
 app.post('/posts/:post_id/comments', function (req, res) {
-	var isOk = true,
-		post = db.posts[req.params.post_id],
-		comment_number;
-	
-	if (req.query.content === undefined) {
-		error.errorMessage += "Nincsen a commentnek tartalma!\n";
-		isOk = false;
-	} 
-	if (req.query.posted_by === undefined) {
-		error.errorMessage += "Név nélkül nem lehet posztolni!\n";
-		isOk = false;
-	}
-	if (!isOk) {
-		res.send(error);
-	} 
-	
-	comment_number = ++ db.posts[req.params.post_id].comments.lastCommentId;
-					
-	post.comments[comment_number] = {
-		post_id: req.params.post_id, //a felso post id
-		id:  comment_number, //random szam,
-		content: req.query.content, //post-oltam,
-		created_at: new Date(), //tudod mit kell itt csinal
-		posted_by: req.query.posted_by //post-oltam
-	};
-				
-	res.send(post.comments[comment_number]); 
-	
+	Comment.create({ author : req.query.author, content : req.query.content, post_id : req.params.post_id }, function (err, comment){
+		var errmessage = {};
+		if (err) {
+			if (err.errors.content){
+				errmessage.content = err.errors.content.message;
+			}
+			if (err.errors.author) {
+				errmessage.author = err.errors.author.message;
+			}
+			res.send(errmessage);
+		} else {
+			res.send(comment);
+		}	
+	}) 
 });
 
 app.put('/posts/:post_id', function (req, res) {
-	var post = db.posts[req.params.post_id];
-	
-	if (post === undefined) {
-		error.errorMessage = 'Nincs ilyen post!';
-		res.send(error);
+	var data = {};
+	if (req.query.content !== undefined) {
+		data.content = req.query.content;
+	}
+	if (req.query.title !== undefined) {
+		data.title = req.query.title;
+	} else {
+		Post.findOneAndUpdate({ _id : req.params.post_id },data, function (err, post){
+			if (err) {
+				res.send('Nincs ilyen post!');
+			} else {
+				res.send(post);
+			}
+		})
 	}
-	
-	post.content = req.query.content,
-	post.title = req.query.title;
-	res.send(post);
-	
 });
 
 app.put('/posts/:post_id/comments/:comment_id', function (req, res) {
-	var post = db.posts[req.params.post_id],
-		comment = post.comments[req.params.comment_id],
-		isOk = true;
-	
-	error.errorMessage = '';
-	
-	if (post === undefined || comment === undefined) {
-		error.errorMessage += 'Nincs módosítani való!';
-		return res.send(error);
-	}	
-	if (!req.query.content) {
-		error.errorMessage += 'Nincs tartalom!';
-		isOk = false;			
+	var data = {};
+	if (req.query.content !== undefined) {
+		data.content = req.query.content;
 	}
-	if (!req.query.posted_by) {
-		error.errorMessage += 'Nincs szerzo!';		
-		isOk = false;			
-	}
-	
-	if (!isOk) {
-		return res.send(error);
-	}
-	
-	if (comment.content === req.query.content) {
-		return res.send(comment);
-	}
-	
-	comment.content = req.query.content;
-	res.send(comment);
-	
+	Comment.findOneAndUpdate({ _id : req.params.comment_id, post_id : req.params.post_id }, data, function (err, comment){
```

Comment:
You are trying to update the comment if the data is empty.
No need for the data variable here, it can be solved with a simple if/else